src/components/publish-site-button.tsx

```diff
@@ -1,53 +1,75 @@
 import { cloudUpload } from '@wordpress/icons';
 import { useI18n } from '@wordpress/react-i18n';
-import Button from 'src/components/button';
-import { Tooltip } from 'src/components/tooltip';
+import { useCallback } from 'react';
 import { useSyncSites } from 'src/hooks/sync-sites';
 import { useAuth } from 'src/hooks/use-auth';
 import { useContentTabs } from 'src/hooks/use-content-tabs';
 import { useSiteDetails } from 'src/hooks/use-site-details';
+import { ConnectButton } from 'src/modules/sync/components/connect-button';
 import { useAppDispatch } from 'src/stores';
 import {
 	connectedSitesActions,
 	useGetConnectedSitesForLocalSiteQuery,
 } from 'src/stores/sync/connected-sites';
+import { useGetWpComSitesQuery } from 'src/stores/sync/wpcom-sites';
 
-export const PublishSiteButton = () => {
+export const PublishSiteButton = ( {
+	redirectToSync = true,
+}: {
+	redirectToSync?: boolean;
+} = {} ) => {
 	const { __ } = useI18n();
 	const dispatch = useAppDispatch();
 	const { setSelectedTab } = useContentTabs();
-	const { user } = useAuth();
+	const { user, isAuthenticated } = useAuth();
 	const { selectedSite } = useSiteDetails();
 	const { data: connectedSites = [] } = useGetConnectedSitesForLocalSiteQuery( {
 		localSiteId: selectedSite?.id,
 		userId: user?.id,
 	} );
 	const { isAnySitePulling, isAnySitePushing } = useSyncSites();
+	const connectedSiteIds = connectedSites.map( ( { id } ) => id );
+	const { isUninitialized: isUninitializedSyncSites, refetch: refetchWpComSites } =
+		useGetWpComSitesQuery( { connectedSiteIds, userId: user?.id } );
 	const isAnySiteSyncing = isAnySitePulling || isAnySitePushing;
-	const handlePublishClick = () => {
-		setSelectedTab( 'sync' );
+
+	const handlePublishClick = useCallback( () => {
+		if ( redirectToSync ) {
+			setSelectedTab( 'sync' );
+		}
+		if ( isAuthenticated && ! isUninitializedSyncSites ) {
+			// Refetch sites in the background but ignore errors
+			void refetchWpComSites();
+		}
 		dispatch( connectedSitesActions.openModal( 'push' ) );
-	};
```

Comment:
Why do we need to call `refetchWpComSites` here? If the goal is to refetch whenever the sync modal opens, would it not make more sense to call this from within the sync modal?

Reply:
After doing the changes, it's clear that it was better to do it in the component, thanks for the suggestion! I have done it as part of [5079edf](https://github.com/Automattic/studio/pull/2132/commits/5079edfa49fdd135682b17c5e4e2308657ee7e33)